binary_and_decimal.rb

```diff
@@ -2,9 +2,19 @@
 #    Fill the array randomly with 0’s and 1’s. Print out the array so that it
 #    appears as a binary number. Calculate the decimal value for this binary
 #    number using the algorithm you devised in class. Print out the decimal value.
-def binary_to_decimal
-  puts "Not implemented"
+def binary_to_decimal(arr)
+  exponent = 0
+  result = 0
+  arr.reverse.each do |s|
+    result += s * (2 ** exponent)
+    exponent += 1
+  end
+  return result
+  puts result
 end
```

Comment:
Minor: the problem statement asks you to print the binary and decimal numbers. You are just missing the puts statements. The method need not return anything.